src/gateway/WebSocket.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-case-declarations */
+import WebSocket from 'ws';
+import fetch from 'node-fetch';
+import { GATEWAY_OPCODES, GATEWAY_CLOSE_CODES, EVENTS, Heartbeat, Identify, Payload } from './constants';
+import { client } from '../Client';
+
+export default async function Socket(token: string, intents: number) {
+  const res = await fetch('https://discord.com/api/v8/gateway/bot', {
+    headers: {
+      Authorization: `Bot ${token}`,
+    },
+  });
+  const json = await res.json();
+
+  const ws = new WebSocket(json.url);
+
+  function heartbeat(ms: number) {

```

Comment:
While it can makes sense to declare this function within the Socket function, I don't know if it's necessary and I would probably hoist it. It's not a closure and it doesn't seem necessary to have it there? This is up for discussion as well